### ISSUE TITLE
Compilation under MINGW

### ISSUE THREAD

**Issue by reneparis** [RA-Consulting-GmbH/openscenario.api.test#155]
We use the API within our OpenScenarioEngine (see [here](https://gitlab.eclipse.org/eclipse/simopenpass/openscenario1_engine)) and one of our goals is to provide the executable under windows without the need for MSVC. Therefore it would be great if sources would be compatible to MINGW64 (MSYS2).

What we already know:

1) cmake files do not consider mingw
2) preprocessor-switches prefer WIN32 over "all other" OS', which makes MinGW integration hard
3) visiblity of exported symbols seems to be stricter than under MSVC
4) due to (3) ANTR4 needs to be updated from 4.8 to 4.10, which needs C++17 → we now compile everything with C++17
5) the cmake files prevent ANTR from being compiled under C++17
6) the UtfCpp library needs to be upgraded to support C++17 (3.2.1 works)

Once we've got that through, we'll prepare a PR.

**Comment by reneparis** [comment]
I've finally managed to compile everything with MinGW64 using C++17 and `gcc 12.1.0` (for more details see https://github.com/RA-Consulting-GmbH/openscenario.api.test/pull/156).

I changed minor stuff in the generated files. Of course, we would also have to put that into the groovy templates.
What took me another day was a glitch within [EnumerationsV1_0.h](https://github.com/RA-Consulting-GmbH/openscenario.api.test/pull/156/commits/3f177ab3e7b59f8ad9af09e252bcc734c5968492#diff-99abc71c3bd7ed520c979496f9f8aa16922b42d6ca68a46d7634a8940b9e5945).
I don't know from where this is coming from, but for this file the preprocessor variables ABSOLUTE and RELATIVE are defined, which made it impossible to compile enums using ABSOLUTE and RELATIVE as fields.

After hacking my way through that, everything now compiles and links. I still miss the possibility to define an INSTALL_PREFIX_PATH, so I don't have to copy everything after building.
